sound/hda/hdac_stream.c

```diff
@@ -142,6 +142,22 @@ void snd_hdac_stream_stop(struct hdac_stream *azx_dev)
 }
 EXPORT_SYMBOL_GPL(snd_hdac_stream_stop);
 
+/**
+ * snd_hdac_stop_all_streams - stop all streams if running
+ * @bus: HD-audio core bus
+ */
+void snd_hdac_stop_all_streams(struct hdac_bus *bus)
```

Comment:
Is it necessary to _emphasize_ that it is stopping **all** streams?
As a side not: `snd_hdac_ext_stop_streams()` is stopping the streams (like `azx_stop_all_streams()` does) **and** stops the chips (with `snd_hdac_bus_stop_chip()`, via `azx_stop_chip()`) in one function..
The "azx" path does this for Intel:
```c
if (bus->chip_init) {
	azx_clear_irq_pending(chip);
	azx_stop_all_streams(chip);
	azx_stop_chip(chip);
}
```
While I try to stay away from long function names, I would tempted to note this detail:
```c
void snd_hdac_stop_streams_and_chip(struct hdac_bus *bus)
```

Reply:
Oh man, this is quite the can of worms.

This function is currently only used by the old SKL driver and if you look at how it's using it, it doesn't make sense. E.g. skl_shutdown() calls snd_hdac_ext_stop_streams() and then follows with a call to snd_hdac_bus_stop_chip().

So the only user of this function is not really using it right.

Oh well, I'd vote for for explicit "stop" in the name. It would avoid the kind of misuse that already has mislead the only user of this function.

Reply:
@ujfalusi initially I used 'stop_streams', it was @ranj063 that I use 'stop_all_streams'.

I did not click on the stop_chip, but indeed it may be worth calling out what happens in that function, so that other users can fix their code. Good suggestion.

@kv2019i the only user of that function was confused by their own creation. haha what a mess...